pkg/crypto11/common.go

```diff
@@ -0,0 +1,23 @@
+package crypto11
+
+import "C"
+import (
+	"unsafe"
+)
+
+func ulongToBytes(n uint) []byte {
+	return C.GoBytes(unsafe.Pointer(&n), C.sizeof_ulong) // ugh!
+}
+
+func concat(slices ...[]byte) []byte {
+	n := 0
+	for _, slice := range slices {
+		n += len(slice)
+	}
+	r := make([]byte, n)
+	n = 0
+	for _, slice := range slices {
+		n += copy(r[n:], slice)
+	}
+	return r
+}
```

Comment:
why are we doing this instead of a simple append?

Reply:
This coding copied one to one from the original project

Reply:
Yeah, I see that, was just wondering if this is really necessary...but if it's copied over I guess that's fine. But in that case, shouldn't the file have some kind of header, such as the rsa.go file?